paasta_tools/monitoring/check_mesos_outdated_tasks.py

```diff
@@ -0,0 +1,122 @@
+
+
+def marathon_tasks(state):
+    for framework in state.get('frameworks', []):
+        if framework['name'].lower().startswith('marathon'):
+            for task in framework.get('tasks', []):
+                yield task
+
+
+def create_slave_id_to_hostname_dict(state):
+    res = {}
+    for slave in state['slaves']:
+        res[slave['id']] = slave['hostname']
+    return res
+
+
+def group_running_tasks_by_job_id_and_gitsha(state):
+    res = {}
+    for t in marathon_tasks(state):
+        if t['state'] == 'TASK_RUNNING':
+            job_id = t['name'][:t['name'].find('.', t['name'].find('.') + 1)]
+            gitsha = t['name'][len(job_id) + 1:t['name'].find('.', len(job_id) + 1)]
+            res.setdefault(job_id, {}).setdefault(gitsha, []).append(t)
+    return res
+
+
+def detect_outdated_gitshas(versions, max_bounce_time=4 * 3600):
+    """Find versions that should have drained more than bounce_time ago (in seconds)"""
+    deploy_time = {}
+    res = []
+    if len(versions) > 1:
+        latest_deploy = 0
+        for version, tasks in versions.items():
+            deploy_time[version] = sum(t['statuses'][0]['timestamp'] for t in tasks) / len(tasks)
+            if deploy_time[version] > latest_deploy and time.time() - deploy_time[version] > max_bounce_time:
+                latest_deploy = deploy_time[version]
+        for version, dtime in deploy_time.items():
+            if dtime + max_bounce_time < latest_deploy:
+                res.append(version)
+    return res
+
+
+def report_outdated_instances(job_id, gitsha, tasks, slave_id2hostname):
+    output = []
+    for t in tasks:
+        deploy_time = datetime.datetime.fromtimestamp(int(t['statuses'][0]['timestamp'])).strftime('%Y-%m-%d %H:%M:%S')
+        container_name = "{}.{}".format(
+            t['slave_id'],
+            t['statuses'][0]['container_status']['container_id']['value'],
+        )
+        hostname = slave_id2hostname[t['slave_id']]
+        hostname = hostname[:hostname.find('.')]
+        output.append(
+            OUTPUT_FORMAT.format(
+                job_id.replace('--', '_')[:30],
+                gitsha[3:],
+                deploy_time,
+                hostname,
+                container_name,
+            ),
+        )
+    return output
+
+
+def check_mesos_tasks():
+    output = []
+    state = get_mesos_state()
+    aggregated_tasks = group_running_tasks_by_job_id_and_gitsha(state)
+    slave_id2hostname = create_slave_id_to_hostname_dict(state)
+    for job_id, versions in aggregated_tasks.items():
+        for gitsha in detect_outdated_gitshas(versions):
+            output.extend(report_outdated_instances(
+                job_id, gitsha, versions[gitsha],
+                slave_id2hostname,
+            ))
+    return output
+
+
+def main():
+    cluster = load_system_paasta_config().get_cluster()
+    output = check_mesos_tasks()
+    if output:
+        print("CRITICAL - There are {} outdated instances running in {}"
```

Comment:
Can you make the 4*3600 a argument so that you can get to it here and print it? (there are N tasks running that are T(time) older than their last bounce)